enclave_build/src/lib.rs

```diff
@@ -58,8 +63,15 @@ pub enum Docker2EifError {
 }
 
 impl<'a> Docker2Eif<'a> {
+    /// If the 'docker_dir' argument is Some (i.e. --docker-dir flag is used), the docker daemon will
+    /// always be used to build the image locally and store it in the docker cache.
+    ///
+    /// If the 'oci_image' argument is Some (i.e. the --image-uri flag is used) and the '--docker-dir'
+    /// flag is not used, pull and cache the image without using the docker daemon.
     pub fn new(
```

Comment:
Wow, this function has a lot of parameters. It is a good practice to limit the number of them to 3-4. Isn't clippy complaining about this?

Reply:
Clippy is not complaining, but this is the original function, I added just 2 new parameters (docker_dir and oci_image).

Reply:
Clippy is actually complaining, but it was silenced by writing allow on line 3 of this file. Since your parameters are optional anyway you can avoid introducing them in the constructor and use a more idiomatic way by initializing them to None when creating the structure and have separate setters for them that can be called only if they're need it.

BTW, directories and paths should not have the type `String`, but should use `Path` or `PathBuf` instead. Paths do not follow the same requirements as strings do.

Reply:
I created structures for grouping the parameters before passing them. There are too few optional fields to make for a builder pattern.
For now I left the paths as Strings as they are represented like this throughout the whole repo. It might be the focus of another PR to propagate `PathBufs`.

Reply:
When there are changes that are not necessarily related to the scope of the PR (and in this case also made by another person) such as this refactoring it's nice to add a separate commit for that.

Reply:
Will make separate initial PR.